zubhub_frontend/zubhub/src/components/upload_file/uploadFile.jsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState, useEffect } from 'react';
+import clsx from 'clsx';
+import ImageIcon from '@material-ui/icons/Image';
+import CustomButton from '../button/Button';
+import {
+  handleFileButtonClick,
+  handleImageFieldChange,
+} from './uploadFileScripts.js';
+import { Typography, FormControl, FormHelperText } from '@material-ui/core';
+import { getFieldAndIndex } from '../../assets/js/utils/scripts';
+
+function UploadFile(props) {
+  const {
+    id,
+    fileType,
+    uploadButtonLabel,
+    classes,
+    countFilesText,
+    multiple,
+    wraperState,
+  } = props;
+
+  const { field, index } = getFieldAndIndex(id);
+  const uploadFileRefs = {
+    fileInput: React.useRef(null),
+  };
+
+  const [state, setState] = useState({
+    media_upload: {
+      upload_dialog: false,
+      images_to_upload: [],
+      videos_to_upload: [],
+      upload_info: {},
+      upload_percent: 0,
+      uploaded_images_url: [],
+      uploaded_videos_url: [],
+    },
+  });
+
+  const handleSetUploadFileState = obj => {
+    if (obj) {
+      Promise.resolve(obj).then(obj => {
+        setState(state => ({ ...state, ...obj }));
+      });
+    }
+  };
+  
+  props = {
+    ...props,
+    UploadFilestate: state,
+    handleSetUploadFileState: handleSetUploadFileState,
+  };
+  return (
+    <div>
+      <CustomButton
+        variant="outlined"
+        size="large"
+        margin="normal"
+        id={[id, 'Button'].join('')}
```

Comment:
for something like this it can be simply
```suggestion
        id={`${id}Button`}
```